0.37.0 fails to build on Alpine Linux 3.9

Thanks for taking the time for reporting an issue!

Before reporting an issue on dd-trace-py, please be sure to provide all
necessary information.

If you're hitting a bug, make sure that you're using the latest version of this
library.

### Which version of dd-trace-py are you using?
0.37.0
### Which version of the libraries are you using?

This was building a wheel of ddtrace only
You can copy/paste the output of `pip freeze` here.

### How can we reproduce your problem?
pip wheel ddtrace==0.37.0
### What is the result that you get?
```
 gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Os -fomit-frame-pointer -g -Os -fomit-frame-pointer -g -Os -fomit-frame-pointer -g -DTHREAD_STACK_SIZE=0x100
000 -fPIC -I/usr/include/python3.6m -c ddtrace/profiling/collector/stack.c -o build/temp.linux-x86_64-3.6/ddtrace/profiling/collector/stack.o -DPy_BUILD_CORE
  ddtrace/profiling/collector/stack.c: In function '__Pyx_PyInt_As_pthread_t':
  ddtrace/profiling/collector/stack.c:8505:58: error: arithmetic on pointer to an incomplete type
       const pthread_t neg_one = (pthread_t) ((pthread_t) 0 - (pthread_t) 1), const_zero = (pthread_t) 0;
                                                            ^
  ddtrace/profiling/collector/stack.c:8428:47: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
               if (unlikely(value != (func_type) (target_type) value)) {\
                                                 ^
  ddtrace/profiling/collector/stack.c:808:43: note: in definition of macro 'unlikely'
     #define unlikely(x) __builtin_expect(!!(x), 0)
                                             ^
  ddtrace/profiling/collector/stack.c:8421:5: note: in expansion of macro '__PYX__VERIFY_RETURN_INT'
       __PYX__VERIFY_RETURN_INT(target_type, func_type, func_value, 0)
       ^~~~~~~~~~~~~~~~~~~~~~~~
  ddtrace/profiling/collector/stack.c:8526:26: note: in expansion of macro '__PYX_VERIFY_RETURN_INT'
                   case  1: __PYX_VERIFY_RETURN_INT(pthread_t, digit, digits[0])
```
### What is the result that you expected?
That it builds.

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -7,6 +7,7 @@ s3_bucket: &s3_bucket pypi.datadoghq.com
 resource_class: &resource_class medium
 busybox_image: &busybox_image busybox:latest
 cimg_base_image: &cimg_base_image cimg/base:2020.01
+python38-alpine_image: &python38-alpine_image python:3.8-alpine
 python38_image: &python38_image circleci/python:3.8
 python37_image: &python37_image circleci/python:3.7
 python36_image: &python36_image circleci/python:3.6
@@ -168,6 +169,10 @@ executors:
     docker:
       - image: *cimg_base_image
     resource_class: *resource_class
+  python38-alpine:
+    docker:
+      - image: *python38-alpine_image
+    resource_class: *resource_class
   python38:
     docker:
       - image: *python38_image
@@ -258,6 +263,11 @@ jobs:
       - run: |
           docker build .
 
+  test_build_alpine:
+    executor: python38-alpine
+    steps:
+      - run: apk add git gcc musl-dev libffi-dev openssl-dev
+      - test_build
   test_build_py38:
     executor: python38
     steps:
@@ -820,6 +830,7 @@ workflows:
       - flake8
 
       # Test building the package
+      - test_build_alpine: *requires_pre_test
       - test_build_py38: *requires_pre_test
       - test_build_py37: *requires_pre_test
       - test_build_py36: *requires_pre_test
@@ -938,6 +949,7 @@ workflows:
             - requestsgevent
             - sqlalchemy
             - sqlite3
+            - test_build_alpine
             - test_build_py38
             - test_build_py37
             - test_build_py36

--- a/ddtrace/profiling/collector/stack.pyx
+++ b/ddtrace/profiling/collector/stack.pyx
@@ -32,8 +32,12 @@ IF UNAME_SYSNAME == "Linux":
 
     cdef extern from "<pthread.h>":
         # POSIX says this might be a struct, but CPython relies on it being an unsigned long.
-        ctypedef unsigned long pthread_t
-        int pthread_getcpuclockid(pthread_t thread, clockid_t *clock_id)
+        # We should be defining pthread_t here like this:
+        # ctypedef unsigned long pthread_t
+        # but e.g. musl libc defines pthread_t as a struct __pthread * which breaks the arithmetic Cython
+        # wants to do.
+        # We pay this with a warning at compilation time, but it works anyhow.
+        int pthread_getcpuclockid(unsigned long thread, clockid_t *clock_id)
 
     cdef p_pthread_getcpuclockid(tid):
         cdef clockid_t clock_id

--- a/docs/installation_quickstart.rst
+++ b/docs/installation_quickstart.rst
@@ -23,11 +23,6 @@ If you want to use the profiler, you'll need to specify the ``profiling`` flavor
 
   $ pip install ddtrace[profiling]
 
-
-.. note::
-
-   The profiler does not work on Alpine Linux.
-
 Quickstart
 ----------
 

